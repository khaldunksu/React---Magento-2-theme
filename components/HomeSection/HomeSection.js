import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import ReactHtmlParser from "react-html-parser";
import "./HomeSection.scss";

export default function HomeSection(props) {
    const { children, classNames, contentClass } = props;

    return (
        <section className={"home-section "+classNames}>
            <div className={"home-section__content "+contentClass}>
                {children}
            </div>
        </section>
    );
}

HomeSection.propTypes = {
    children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
    classNames: PropTypes.string,
    contentClass: PropTypes.string
};
HomeSection.defaultProps = {
    children: {},
    classNames: '',
    contentClass: ''
};

