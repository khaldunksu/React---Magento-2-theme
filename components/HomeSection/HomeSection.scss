.home-section{
  height: 600px;
  width: 100%;
  display: flex;

  &__content{
    display: flex;
  }
}