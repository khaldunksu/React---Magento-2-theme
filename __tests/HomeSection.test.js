import { shallow } from "enzyme";
import React from "react";
import Router from "next/router";
import toJson from "enzyme-to-json";
import HomeSection from "../components/HomeSection/HomeSection";

const mockedRouter = { push: () => {} };
Router.router = mockedRouter;

describe("Component HomeSection matches snapshot", () => {
    it("renders page layout correctly", () => {
        const wrapper = shallow(<HomeSection />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("Component HomeSection", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<HomeSection />);
    });

    it("renders the dumb component", () => {
        expect(wrapper.length).toEqual(1);
    });

});