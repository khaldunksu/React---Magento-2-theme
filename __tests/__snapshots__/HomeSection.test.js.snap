// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component HomeSection matches snapshot renders page layout correctly 1`] = `
<section
  className="home-section "
>
  <div
    className="home-section__content "
  >
    <Component />
  </div>
</section>
`;
